Add documentation for CSV support in the FHIR Mapper (#84)
simplifhir/simplifier-docs: mappingengine/featureslist/supportedformats.rst

All components of the FHIR Mapper are designed to handle customs data formats as the source for a mapping to FHIR.
By default the FHIR Mapper supports reading and writing 'Custom FHIR resources' (:ref:`feature_customresources`), however it is also possible to provide data in other formats besides the JSON / XML serialization format of FHIR. 
The mapping engine therefore supports different 'adapters' which can be used to read in other formats natively.

CSV
-------------

> The FHIR Mapper supports mapping a comma-separated values (CSV) file without any prior pre-processing. You can POST the CSV file as the HTTP Body to the $transform operation and access its metadata and content within a StructureMap.

Reviewer: ```suggestion
The FHIR Mapper supports mapping a comma-separated values (CSV) file without any prior setup. You can POST the CSV file as the HTTP Body to the $transform operation and access its metadata and content within a StructureMap.
```

> To enable CSV support, please adjust the BinaryWrapper settings in you appsettings.instance.json to allow Vonk to accept the ``text/csv`` Content-Type header: ::

Reviewer: ```suggestion
To enable CSV support, please adjust the BinaryWrapper settings in your appsettings.instance.json to allow Vonk to accept the ``text/csv`` Content-Type header: ::
```

      "Vonk.Plugin.BinaryWrapper":{
        "RestrictToMimeTypes": ["application/pdf", "text/plain", "image/png", "image/jpeg", "text/fhir-mapping", "text/csv"]
      }

> In order to fully use the type-safety features of the FHIR Mapping Language, a StructureDefinition is generated automatically by the FHIR Mapper when a CSV file is received. Depending on the composition of the CSV two different formats are used.

Reviewer: ```suggestion
In order to fully use the type-safety features of the FHIR Mapping Language, a StructureDefinition is generated automatically by the FHIR Mapper when a CSV file is received. Depending on the composition of the CSV, two different formats are used:
```

1. CSV file contains a header with column metadata:

For example, given a CSV file that contains the following rows: ::

    ID, Given,        Family,     Gender
    1,  Peter James,  Chalmers,   M
    2,  Sandy,        Notsowell,  F 
    
The following StructureDefinition would be produced in the background while executing $transform: ::

  <embed>

>     <iframe src=https://simplifier.net/embed/render?id=fhirmapperr4/csvtransport height=”345px” width=”100%”></iframe>

Reviewer: These look like smart quotes - might not be what's intended.

  </embed>
  
The .hasHeader child element indicates if the CSV mapping adapter was configured to interpret the first row of the CSV file as a header record. This behaviour can be indicated according to `RFC 4180 <https://tools.ietf.org/html/rfc4180>`_ by using the ``header`` parameter in the Content-Type header.

All other column identifiers are exposed as child elements of the .record BackboneElement. They can directly be used in a mapping rule: ::

  src.record as record then 
  {
    record.ID as id -> tgt.identifier = id('<system>', id);
    record.Given as given -> tgt.name as name, name.given = given;
    record.Family as family -> tgt.name as name collate, name.family = family;
    record.Gener as gender -> tgt.gender = translate('<ConceptMap>', gender, 'code');
  };
  
2. CSV file contains no header metadata:

> Regardless if the CSV file contains a header, all elements are accessible for the mapping to FHIR. For files not containing a metadata row, element names for the mapping source are generated dynamically: :: 

Reviewer: ```suggestion
Regardless if the CSV file contains a header, all elements are accessible for the mapping to FHIR. For files not containing a metadata row, element names for the mapping source are generated dynamically as ``field#``: :: 
```
